WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2909,16 +2916,36 @@ static GSourceFuncs _handlerIntervention =
                     webkit_memory_pressure_settings_free(memoryPressureSettings);
                 }
 #endif
+
+#if WEBKIT_CHECK_VERSION(2, 42, 0)
+                double originStorageRatio = -1.0;    // -1.0 means WebKit will use the default quota (1GB)
+                if (_config.IndexedDBSize.IsSet() && _config.IndexedDBSize.Value() != 0) {
+                    originStorageRatio = static_cast<double>(_config.IndexedDBSize.Value());
+                }
+
+                double totalStorageRatio = -1.0;    // -1.0 means there's no limit for the total storage
+                if (_config.IndexedDBTotalSize.IsSet() && _config.IndexedDBTotalSize.Value() != 0) {
+                    totalStorageRatio = static_cast<double>(_config.IndexedDBTotalSize.Value());
+                }
+
                 auto* websiteDataManager = webkit_website_data_manager_new(
                     "local-storage-directory", wpeStoragePath,
                     "disk-cache-directory", wpeDiskCachePath,
                     "local-storage-quota", localStorageDatabaseQuotaInBytes,
                     "indexeddb-directory", indexedDBPath,
-                    "per-origin-storage-quota", indexedDBSizeBytes,
+                    "origin-storage-ratio", originStorageRatio,
+                    "total-storage-ratio", totalStorageRatio,
+                     nullptr);
+                g_free(indexedDBPath);
+#else
+                auto* websiteDataManager = webkit_website_data_manager_new(
+                    "local-storage-directory", wpeStoragePath,
+                    "disk-cache-directory", wpeDiskCachePath,
+                    "local-storage-quota", localStorageDatabaseQuotaInBytes,
                      nullptr);
+#endif
                 g_free(wpeStoragePath);
                 g_free(wpeDiskCachePath);
-                g_free(indexedDBPath);
```

Comment:
I think you now have a leek in the < 2.42 situation... The g_free was moved inside the #if WEBKIT_CHECK_VERSION(2, 42, 0) but it seems indexedDBPath is also created in the older version case.

Reply:
You're right, I think I forgot to to add the whole `indexedDBPath` section above inside `#if WEBKIT_CHECK_VERSION(2, 42, 0)` after we removed the support of `IndexedDB` for previous WebKit API versions.